### DISCUSSION TITLE
[week 16] 第十六週練習題

### DISCUSSION THREAD

**Discussion by Jacksonnnnnnnnnnnn** [Lidemy/forum#180] [程式導師實驗計畫] [answered]
> Q1、請你寫出一個 Robot 的 class，初始化的時候可以設置座標 x 跟 y
接著 Robot 會有兩個方法，getCurrentPosition 跟 go，前者會回傳現在機器人所在的 x 與 y 座標，後者可以讓機器人往東南西北任一方向移動，需要傳進 'N', "E", 'S', 'W' 任何一個字串，代表要往哪一個方向走…往北走 Y 座標會增加，往南走 Y 座標會減少，往東走 X 座標會增加，往西走 X 座標會減少

function 執行內容寫在註解裡：
```js
export class Robot {
// 使用 constructor 引入坐標 x 跟 y
  constructor(x, y) {
    this.x = x;
    this.y = y;
  }

// 回傳現在機器人所在的 x 與 y 座標
  getCurrentPosition() {
    return {
      x: this.x,
      y: this.y
    };
  }

// 機器人往東南西北任一方向移動，需要傳進 'N', "E", 'S', 'W' 任何一個字串，'N' 的話 y 增加，'S' 的話 y 減少，'E' 的話 x 增加，'W' 的話 X 減少
  go(direction) {
    switch (direction) {
      case "N":
        this.y += 1;
        this.getCurrentPosition();
        break;
      case "S":
        this.y -= 1;
        this.getCurrentPosition();
        break;
      case "E":
        this.x += 1;
        this.getCurrentPosition();
        break;
      case "W":
        this.x -= 1;
        this.getCurrentPosition();
        break;
      default:
        console.log("please input N or E or S or W");
    }
  }
}

const robot = new Robot(10, 10);
```
error message
> expect(received).toEqual(expected) // deep equality
Expected: {"x": 10, "y": 10}
Received: undefined

![Imgur](https://imgur.com/tKGKmPh.png)
> *備註，圖中的另外兩個 problems 是 Q2, Q3，因為我還沒做。

依照錯誤信息，是驗證系統無法接收我的值嗎？ 請大大們相助，感謝。

**Comment by aszx87410** [comment, marked as answer]
請附上可重現這個狀況的 codesandbox 網址

**Reply by Jacksonnnnnnnnnnnn** [reply]
謝謝老師提醒，剛把網址拿去無痕視窗測試，發現顯示 passed！再重刷本來的頁面也顯示 passed。不確定是否在 codesandbox 的頁面上太久，系統沒抓到 ?!
https://codesandbox.io/s/week16-homeworks-forked-95hxc?file=/index.js
